src/include/parser/expression/abstract_expression.h

```diff
@@ -215,48 +213,45 @@ class AbstractExpression {
   virtual void FromJson(const nlohmann::json &j);
 
  private:
-  /**
-   * Type of the current expression
-   */
+  /** Type of the current expression */
   ExpressionType expression_type_;
-
-  /**
-   * Name of the current expression
-   */
+  /** Name of the current expression */
   std::string expression_name_;
-  /**
-   * Alias of the current expression
-   */
+  /** Alias of the current expression */
   std::string alias_;
-  /**
-   * Type of the return value
-   */
+  /** Type of the return value */
   type::TypeId return_value_type_;
 
   /**
-   * The current sub-query depth level in the current expression, -1
-   *  stands for not derived
+   * MUTABLE Sub-query depth level for the current expression.
+   *
+   * TODO(WAN): ask LING to document her assumptions, I see that depth can still be -1 after calling DeriveDepth().
+   *
+   * DeriveDepth() MUST be called on this expression tree whenever the structure of the tree is modified.
+   * -1 indicates that the depth has not been set, but we have no safeguard for maintaining accurate depths between
+   * tree modifications.
    */
   int depth_ = -1;
   /**
-   * The flag indicating if there's sub-query in the current expression
-   */
+   * MUTABLE Flag indicating if there's a sub-query in the current expression or in any of its children.
+   *
+   * TODO(WAN): check with LING on why we need this and whether we can roll DeriveSubqueryFlag into DeriveDepth.
+   * */
```

Comment:
has_subquery: 
this is essentially use to detect if the query predicate IsSupportedConjunctivePredicate ; according to Peloton comment:
```
 // Currently support : 1. expr without subquery
  // 2. subquery without disjunction. Since the expr is already one of the
  // conjunctive exprs, we'll only need to check if the root level is an
  // operator with subquery
```
I wonder if that is true; we might be able to remove the check if we get everything supported

Depth:
according to Peloton comment:
```
// Depth is used to detect correlated subquery, it is set in the binder,
    // if a TVE has depth less than the depth of the current operator, then it is correlated predicate
 ```